### ISSUE TITLE
fix a litle bit to match as uvloop in fastapi

### ISSUE THREAD

**Issue by Khanhlinhdang** [Vizonex/Winloop#35]
add arg "use_subprocess" on function install contained in file __iniit__.py and then we add winloop to this config of fastapi, so we can use normal like uvloop in fastapi: 

uvicorn.config.LOOP_SETUPS.update({"winloop":"winloop:install"})

see the code to understand what I am taking about.

def install(use_subprocess=None) -> None:
    """A helper function to install winloop policy."""
    if _sys.version_info[:2] >= (3, 12):
        _warnings.warn(
            'winloop.install() is deprecated in favor of winloop.run()'
            'starting with Python 3.12.',
            DeprecationWarning,
            stacklevel=1,
        )
    __asyncio.set_event_loop_policy(EventLoopPolicy())

`
# TODO this code example is deprecated
from fastapi import FastAPI
from fastapi.responses import HTMLResponse
import winloop
import uvicorn
import asyncio
import datetime

app = FastAPI()

@app.get("/test")
async def test_get_request():
    return HTMLResponse("<html><body><h1>FAST API WORKS WITH WINLOOP!</h1></body></html>")

# starllete will use asyncio.to_thread() so that this can remain asynchronous
@app.get("/date")
def test_dynamic_response():
    return str(datetime.datetime.now())

# Although tricky to pass and is not normal, it does in fact work...
if __name__ == "__main__":
    uvicorn.config.LOOP_SETUPS.update({"winloop":"winloop:install"})
    # Winloop's eventlooppolicy will be passed to uvicorn after this point...
    config = uvicorn.Config(app=app,host="localhost",port=2022,loop="winloop")
    server = uvicorn.Server(config)
    server.run()
`

**Comment by lschoe** [comment]
Looks like you can arrange this yourself by adding next to your code above a local file `mywinloop.py` with content
```python
import asyncio
import winloop

def winloop_setup(use_subprocess: bool = False) -> None:
    asyncio.set_event_loop_policy(winloop.EventLoopPolicy())
```
and then use it in your above code using
```python
uvicorn.config.LOOP_SETUPS.update({"winloop":"mywinloop:winloop_setup"})
```
instead?

**Comment by Khanhlinhdang** [comment]
Exactly, that's what i did

**Comment by Khanhlinhdang** [comment]
`from fastapi import FastAPI
from fastapi.responses import HTMLResponse
import uvicorn
import datetime

app = FastAPI()

@app.get("/test")
async def test_get_request():
  return HTMLResponse("FAST API WORKS WITH WINLOOP!")

@app.get("/date")
def test_dynamic_response():
  return str(datetime.datetime.now())

if __name__ == "main":
  # Winloop's eventlooppolicy will be passed to uvicorn after this point...
  uvicorn.config.LOOP_SETUPS.update({"winloop":"winloop:install"})
  config = uvicorn.Config(app=app,host="localhost",port=2022,loop="winloop")
  server = uvicorn.Server(config)
  server.run()`

**Comment by Khanhlinhdang** [comment]
example
![image](https://github.com/user-attachments/assets/5d1feeb1-468b-428b-8417-b002aaf30d98)

**Comment by lschoe** [comment]
Hi @Khanhlinhdang, right, your first comment "Exactly, that's what i did" was overwritten above, which may be helpful to other readers to repeat here.

My point is that you can follow your setup to make `uvicorn` work with winloop, which is nice. But then without modifying winloop (hence without modifying `install()`). Instead, creating the auxiliary local module `mywinloop` as shown above. This follows the approach adopted by `uvicorn`, see https://github.com/encode/uvicorn/tree/master/uvicorn/loops.

**Comment by Khanhlinhdang** [comment]
Yes, I agree with you. use module mywinloop above to do not modify the origin winloop.
